crates/core/tedge_api/src/parser.rs

```diff
@@ -361,4 +362,23 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn parse_wrong_type() -> anyhow::Result<()> {
```

Comment:
The name of the test is misleading and fails to convey that *now* a measurement can be named "type".

I would improve this test by checking the value `456` is attached to a measurement named type.

I would also a test case with `"type": "456"` checking that this is not error and that the value is simply ignore.

These two tests will mostly have a documentation purpose as this behavior change is not obvious.

Reply:
I have a mixed opinion about whether to have this `type` of checking here. Because we can parse any `property` with `text` value, not just restricted to `type/externalSource`.  I feel it's better to have a test in `c8y_mapper_ext`.

Reply:
> Because we can parse any property with text value, not just restricted to type/externalSource. 

Such a test would be useful only as an active documentation to tell that "type" and "externalSource" have a special meaning for C8Y but not for thin-edge.